python_modules/dagster/dagster/_scheduler/scheduler.py

```diff
@@ -282,46 +286,72 @@ def launch_scheduled_runs(
 
     tick_retention_settings = instance.get_tick_retention_settings(InstigatorType.SCHEDULE)
 
-    schedules: Dict[str, ExternalSchedule] = {}
+    running_schedules: Dict[str, ExternalSchedule] = {}
+    all_workspace_schedule_selector_ids = set()
     error_locations = set()
 
+    now_timestamp = end_datetime_utc.timestamp()
+
     for location_entry in workspace_snapshot.values():
         code_location = location_entry.code_location
         if code_location:
             for repo in code_location.get_repositories().values():
                 for schedule in repo.get_external_schedules():
                     selector_id = schedule.selector_id
+                    all_workspace_schedule_selector_ids.add(selector_id)
                     if schedule.get_current_instigator_state(
                         all_schedule_states.get(selector_id)
                     ).is_running:
-                        schedules[selector_id] = schedule
+                        running_schedules[selector_id] = schedule
+                    elif all_schedule_states.get(selector_id):
+                        schedule_state = all_schedule_states[selector_id]
+                        # If there is a DB row to update, see if we should still update the
+                        # last_iteration_timestamp
+                        _write_and_get_next_checkpoint_timestamp(
```

Comment:
just confirming that there is a test that would fail if this were not here, b/c its important for things to work correctly when they are in a STOPPED state right?

Reply:
Yeah, I had to stack on top of your change so that tests would pass.